### PR TITLE
fix #136

### LINKED ISSUE #136: Nagios 4.4.0+ incompatibility
@sni do you want me to try and get Nagios Core 4.4.0+ working with mod_gearman? I've identified the issues and am working on it internally - I wouldn't mind submitting this as a PR when complete.

### PR DESCRIPTION
**Pull request by amuhametov**
need someone to verify it works

### CODE REVIEW

**Comment by sni** [comment]
i don't have any nagios 4 setups, but i am willing to merge anything which doesn't break the naemon or icinga neb module :-)

**Comment by amuhametov** [comment]
@sni do you mind I add neb_module_nagios44 to make it possible to use same mod_gearman package with both nagios 4.3 and 4.4?

**Comment by sni** [comment]
No, i don't want that. It would be much easier if there would be a nagios4-devel package containing the headers and library. Then you could just use the version you like. Thats the way we do it with naemon.

File: neb_module/mod_gearman.c [hunk 1]
```diff
@@ -982,7 +982,7 @@ static int handle_host_check( int event_type, void *data ) {
 #ifdef USENAGIOS3
     adjust_host_check_attempt_3x(hst,TRUE);
 #endif
-#if defined(USENAEMON) || defined(USENAGIOS4)
+#ifdef USENAEMON
```
Comment (sni) [inline neb_module/mod_gearman.c]: That looks like it would break things.

Comment (amuhametov) [reply neb_module/mod_gearman.c]: What would be the best way?

**Comment by Madlohe** [comment]
I don't think distributing multiple versions of the headers would work here. The issue is that adjust_host_check_attempt() was removed between nagios 4.3.4 and 4.4.0. It does actually still exist in the headers but the definition/calls to it no longer exist. I'll talk to @hedenface about adding back an empty definition for the function but as it stands this works in the most recent version.

**Comment by hedenface** [comment]
I'd rather not have an empty function definition - but we should definitely remove it from the headers.

Why can't we just remove the calls to the function from mg?

**Comment by sni** [comment]
If it was removed on purpose, its ok, just wanted to make sure its not a mistake. Just keep it like that then.

**Comment by amuhametov** [comment]
fix #136 
Added nagios 4.4 support
May be (sure will be) incompatible with nagios-core <= 4.3

**Comment by amuhametov** [comment]
So, after 2 weeks I can say it works fine.

@sni  will you merge this pr?

**Comment by sni** [comment]
sure, thanks a lot

**Comment by amuhametov** [comment]
@Madlohe do you use thruk?

**Comment by Madlohe** [comment]
@amuhametov nope! I'm a developer for Nagios Enterprises (one of @hedenface 's underlings).

**Pull request merged**